src/communication/AsyncUDPClient.cpp

```diff
@@ -32,74 +32,83 @@
  */
 //----------------------------------------------------------------------
 
+#include <sick_safetyscanners_base/logging/logging_wrapper.h>
 
 #include <sick_safetyscanners_base/communication/AsyncUDPClient.h>
 
 namespace sick {
 namespace communication {
 AsyncUDPClient::AsyncUDPClient(const PacketHandler& packet_handler,
-                               boost::asio::io_service& io_service,
-                               const uint16_t& local_port)
+                               boost::asio::io_service& io_service)
   : m_packet_handler(packet_handler)
   , m_io_service(io_service)
 {
-  // Keep io_service busy
-  m_io_work_ptr = std::make_shared<boost::asio::io_service::work>(boost::ref(m_io_service));
   try
   {
-    m_socket_ptr = std::make_shared<boost::asio::ip::udp::socket>(
-      boost::ref(m_io_service),
-      boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), local_port));
+    m_socket_ptr = std::make_shared<boost::asio::ip::udp::socket>(m_io_service);
   }
   catch (std::exception& e)
   {
     ROS_ERROR("Exception while creating socket: %s", e.what());
   }
+
   ROS_INFO("UDP client is setup");
 }
 
 AsyncUDPClient::~AsyncUDPClient()
 {
-  m_io_service.stop();
```

Comment:
This call stops the complete io_service and prevents further use of the io_service in the complete library. This was ok in the case where the io_service was created by the lib, but it will unexpected behavior if the io_service is externally provided.

Reply:
> I did some cleanup with the use of the io_service because it was sometimes not used as proposed in the boost docu